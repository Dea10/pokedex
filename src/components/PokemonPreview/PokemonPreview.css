.pokemon-preview-container {
    display: flex;
    justify-content: center;
}

.pokemon-preview {
    display: flex;
    width: 100px;
    height: 100px;
    border: 1px solid yellow;
    justify-content: center;
    align-items: center;
    border-radius: 8px;
}

.preview-text {
    text-align: center;
}