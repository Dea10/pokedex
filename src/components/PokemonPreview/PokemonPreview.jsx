import './PokemonPreview.css';

const PokemonPreview = ({ pokemonSpriteUrl=null }) => {
    return (
        <div className='pokemon-preview-container'>
            <div className="pokemon-preview">
                {
                    !!pokemonSpriteUrl ?
                        <img src={pokemonSpriteUrl} alt="" /> :
                        <small className='preview-text'>Click on a pokemon to preview</small>
                }
            </div>
        </div>
    )
}

export default PokemonPreview;