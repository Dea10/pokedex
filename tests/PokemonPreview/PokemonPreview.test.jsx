import { render } from "@testing-library/react"
import PokemonPreview from "../../src/components/PokemonPreview/PokemonPreview"

describe('PokemonPreview test', () => { 
    test('should match snapshot with no pokemon redered', () => {
        const {container} = render(<PokemonPreview />)

        expect(container).toMatchSnapshot();
    })

    test('should match snapshot with pokemon redered', () => {
        const pokemonSpriteUrl = 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/132.png'
        const {container} = render(<PokemonPreview pokemonSpriteUrl={pokemonSpriteUrl} />)

        expect(container).toMatchSnapshot();
    })
})