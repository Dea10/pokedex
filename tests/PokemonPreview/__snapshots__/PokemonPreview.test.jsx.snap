// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPreview test should match snapshot with no pokemon redered 1`] = `
<div>
  <div
    class="pokemon-preview-container"
  >
    <div
      class="pokemon-preview"
    >
      <small
        class="preview-text"
      >
        Click on a pokemon to preview
      </small>
    </div>
  </div>
</div>
`;

exports[`PokemonPreview test should match snapshot with pokemon redered 1`] = `
<div>
  <div
    class="pokemon-preview-container"
  >
    <div
      class="pokemon-preview"
    >
      <img
        alt=""
        src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/132.png"
      />
    </div>
  </div>
</div>
`;
